### DISCUSSION TITLE
Ability to run evaluations using prompt from prompt editor with content from spans

### DISCUSSION THREAD

**Discussion by p-c-mo** [langfuse/langfuse#4851] [Ideas]
### Describe the feature or potential improvement

Hi,

I wanted to highlight some functionality I believe would be useful. I have built this myself at a past employer.

Often when I am iterating on a prompt, it is useful to run it against past user inquiries. For example, if I have a RAG tool that needs to categorize a user inquiry at a particular step, it is useful to be able to iterate on different prompts against messages we've received before, and then evaluate based on some criteria.

Once a customer has integrated langfuse, it is essentially a repository for previous user interactions, and it would be wonderful if I could take advantage of messages in generations that have been captured.

I tried to experiment using datasets, but was unable to accomplish this using the current tooling.

**Comment by marcklingen** [comment]
Thanks for sharing! This is a common workflow when using Langfuse Prompt Experiments and datasets, although it is a bit more complicated than it needs to be as it requires to parse the variables inserted into a prompt before creating the datasets used for the prompt experiment. I agree that this could be simplified, let's track interest for this here.
